### ISSUE TITLE
Add .Position to the context of heading and image render hooks

### ISSUE THREAD

**Issue by bowman2001** [gohugoio/hugo#11325]
The codeblock render hook context includes the `.Position` of the current codeblock in a local Markdown file.

It would be nice to have this information also for heading and image render hooks. This would allow to include links to the source in local server mode.

**Comment by bep** [comment]
I agree that this would be nice to have, but it's non-trivial to implement. The implementation we have for code blocks is expensive/slow. The main challenge is the shortcodes with `{{<` delimiters; the position you get when parsing the Markdown (Goldmark, assuming that info is there) does not take the shortcodes into account. And if you think about the new {https://github.com/gohugoio/hugo/pull/11320 ... there's lots to think about, as Position may well point to a different Markdown file.

**Comment by bowman2001** [comment]
I’ve hoped this would be relatively easy. So it can wait, it’s certainly not a priority.

**Comment by jmooring** [comment]
This was implemented in v0.160.0.
See <https://github.com/gohugoio/hugo/pull/14664>.
